### DISCUSSION TITLE
List Interface Improvements

### DISCUSSION THREAD

**Discussion by Kematia** [directus/directus#6635] [Feature Requests]
Hey Gang! I have a couple suggestions for improving the UX of the list interfaces.

## Copy/Cut/Paste/Duplicate
This is the most straightforward. Adds the ability to copy, cut, paste and delete an item in a list interface.

We could take a leaf out of Storyblok, which has a nice and intuitive UI for C/P/C/D

![image](https://user-images.githubusercontent.com/17851386/124344921-c4ba6900-dbd5-11eb-80ed-550f67ee64f6.png)

Cutting and pasting is much easier than dragging an item half a mile 😆 

## Add New Above/Below

Storyblok again has a `Add New` icon on each element, either above or below the current item.

Non-working poc:

https://user-images.githubusercontent.com/17851386/124345871-cd15a280-dbdb-11eb-826d-5a4c09e529ef.mp4

## Delete Confirmation
I feel the current delete action is lacking a confirmation. ***It's way too easy to delete a JSON/relational item.***

 A full fledged modal for deleting a non-relational field would be overkill, so I thought we could go with an inline confirmation:

https://user-images.githubusercontent.com/17851386/124345533-cbe37600-dbd9-11eb-8717-5170961cc820.mp4

A modal confirmation could be used when deleting a relational item.

### Relational data
I reckon implementing the above for JSON data would easy, but some features (like duplication) could get trickier for relation data.

Cut/Paste is doable for both, as it's just array manipulation (JSON) or patching the `sort` field (relational)

***

If we can come up with a design spec I'm willing to work on this! This is something I'd love to implement into Core 😄

**Comment by rijkvanzanten** [comment]
I'm wondering if the add-above/below is enough of a power user improvement to warrant so many extra interface buttons 🤔 

The delete confirmation makes perfect sense, though I'm wondering if it should be a simple extra button there, or if we should rely on a popover. I think that might a little more "on purpose". I'm proposing we use the triple-dot on the right edge, and use a v-menu with duplicate/delete.

**Reply by Kematia** [reply]
I feel like the Cut/Paste buttons are pretty useful. I see my clients struggling with long lists and having only the Create New button at the bottom. Also I feel like dragging is finnicky*, and not the best for less experienced users.

I reckon we could start with the menu + a couple options (delete and duplicate as you said), and then see from there? We may need some design mockups for the rest of the buttons.

_*dragging in general, not the Directus implementation_

**Reply by rijkvanzanten** [reply]
> I reckon we could start with the menu + a couple options (delete and duplicate as you said), and then see from there?

Yeah, that works for me!

**Reply by Kematia** [reply]
![image](https://user-images.githubusercontent.com/17851386/125159949-bd1d3600-e17a-11eb-9a24-38e002fe4018.png)

Well, we do already have this pattern here!

**Reply by rijkvanzanten** [reply]
Yeah! We're using the same pattern all over the place 🙂

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
